core/src/main/java/org/apache/calcite/rel/rules/HyperGraph.java

```diff
@@ -401,69 +453,67 @@ public RexNode extractJoinCond(RelNode left, RelNode right, List<HyperEdge> edge
     };
 
     for (HyperEdge edge : edges) {
-      RexNode inputRefCond = edge.getCondition().accept(inputName2InputRefShuttle);
+      RexNode inputRefCond = edge.getCondition().accept(shuttle);
       joinConds.add(inputRefCond);
     }
-    return RexUtil.composeConjunction(left.getCluster().getRexBuilder(), joinConds);
+    return RexUtil.composeConjunction(getCluster().getRexBuilder(), joinConds);
   }
 
   /**
-   * Before starting enumeration, add Project on every input, make all field name unique.
-   * Convert the HyperEdge condition from RexInputRef to RexInputFieldName
+   * Restore the projection order of the final result to the original plan.
```

Comment:
ensure that the fields produced by the reordered join are in the same order as in the original plan.